internal/resource/resource_test.go

```diff
@@ -81,6 +81,59 @@ var newResourceTests = []struct {
 			assert.Equal(t, 20*time.Second, r.ReconcileInterval.Duration)
 		},
 	},
+	{
+		Name: "replace override",
+		Manifest: `{
+			"apiVersion": "apps/v1",
+			"kind": "Deployment",
+			"metadata": {
+				"name": "foo",
+				"namespace": "bar",
+				"annotations": {
+					"eno.azure.io/overrides": "[{\"path\":\"self.metadata.annotations[\\\"eno.azure.io/replace\\\"]\", \"value\":\"true\"}]"
+				}
+			}
+		}`,
+		Assert: func(t *testing.T, r *Snapshot) {
+			assert.True(t, r.Replace)
+		},
+	},
+	{
+		Name: "conditional replace override, positive",
+		Manifest: `{
+			"apiVersion": "apps/v1",
+			"kind": "Deployment",
+			"metadata": {
+				"name": "foo",
+				"namespace": "bar",
+				"annotations": {
+					"trigger-replace": "yespls",
+					"eno.azure.io/overrides": "[{\"path\":\"self.metadata.annotations[\\\"eno.azure.io/replace\\\"]\", \"value\":\"true\", \"condition\": \"self.metadata.annotations[\\\"trigger-replace\\\"] == 'yespls'\"}]"
```

Comment:
test startsWith? I tried and I guess couldn't get the escaping right

Reply:
The integration test uses it and I did some manual tests too. Think we should add some to `internal/cel`?